app/api/og-image/cast/[username]/[hash]/[[...rank]]/route.tsx

```diff
@@ -0,0 +1,198 @@
+import getUserByUsername from '@/lib/neynar/getNeynarUserByUsername';
+import {
+  getEmbedAndMetadata,
+  getChannelData,
+  formatPoints,
+  replaceSpecialCharacters,
+} from '@/lib/utils';
+import { ImageResponse } from '@vercel/og';
+import { NextRequest } from 'next/server';
+
+export const runtime = 'edge';
+
+export async function GET(
+  req: NextRequest,
+  { params }: { params: { username: string; hash: string; rank: string } },
+) {
+  const { username, hash, rank } = params;
+
+  const encodedUsername = replaceSpecialCharacters(username);
+
+  const userProfile = await getUserByUsername(username);
+  const profilePfp = userProfile?.pfp?.url;
+
+  const { cast, metadata } = await getEmbedAndMetadata(hash);
+  const channelData = getChannelData(cast?.channelId);
+
+  const channelLabel = channelData?.label || '/sonata';
+  const channelIcon = channelData?.icon || 'https://i.imgur.com/Xa4LjYA.jpeg';
+
+  const points = formatPoints(cast?.points);
+
+  const soraSemiBold = fetch(new URL('/public/Sora-SemiBold.ttf', import.meta.url)).then((res) =>
+    res.arrayBuffer(),
+  );
+
+  const soraNormal = fetch(new URL('/public/Sora-Regular.ttf', import.meta.url)).then((res) =>
+    res.arrayBuffer(),
+  );
```

Comment:
move all this logic into standalone libs
- rule: each function should do exactly one thing
- actual: this function is doing `profile lookup`, `metadata lookup`, `channel lookup`, etc all defined in the same file
- solution: move this logic into a single lib, called by this API endpoint